pkg/utils/utils.go

```diff
@@ -402,3 +407,15 @@ func Retry(retries int, sleep time.Duration, f func() error) error {
 	}
 	return err
 }
+
+func warnOnError(callerFn string, f func() error) func() error {
```

Comment:
Let's simplify this. You can simply put the logging in the Retry function.

Reply:
It would involve many log calls, one for each error returned. Here we catch all al them.

I'm ok in removing all this logging and discussing it in another PR

Reply:
but I didn't mean to add it inside the func() which is passed as a parameter in the Retry function, but to add it in the Retry function itself. Line 400 of utils.go. You would only to add it once.

Reply:
Oh, I got it. Not a big fan of logging from utility functions. it can lead to a lot of junk in the log files, for example, if the `Retry` function is called for another job where it's expected to have multiple failures before going well.

since it's not yet used widely, I'd overcome that by making it private

Reply:
you can also add a new parameter "warnOnError" in the Retry func to control when to log

Reply:
5th parameter for a utility function. That's going to be even worse.
Thanks for the suggestions, but let's discuss it in a different PR, we went out of topic here.

Reverting

Reply:
alright, please create an issue to not forget this.